### DISCUSSION TITLE
How do you type the notification option?

### DISCUSSION THREAD

**Discussion by AsuraKev** [ant-design/ant-design#45145] [Q&A] [answered]
I am trying to provide a correct type for the configuration which you can pass to notification instance, but I am getting the following type mismatch. what is the correct type for the notification options? 

![image](https://github.com/ant-design/ant-design/assets/52765572/d40d31c6-dc99-4d58-80be-027fc55a8373)

**Comment by yoyo837** [comment]
Let's look at your `option` definition, show us the code please.

**Reply by AsuraKev** [reply]
NotificationOption is imported from Antd. But .open() expects a ArgProps type

**Reply by kiner-tang** [reply]
It seem that antd did not export NotificationOptions type, Can you provide the code you import it?

**Reply by AsuraKev** [reply]
Sorry its NotificationConfig that I am using, please find below screenshot
![image](https://github.com/ant-design/ant-design/assets/52765572/5e0a172f-7290-4e05-9781-824283538602)

**Reply by kiner-tang** [reply]
<img width="620" alt="image" src="https://github.com/ant-design/ant-design/assets/10286961/79a3b063-3035-41f8-9bca-0683a21e41f2">

`notification.open` need pass params with ArgsProps type, not `NotificationConfig`. You can use like this:

```tsx
notification.open({
  message: "This is test message"
})
```

**Reply by AsuraKev** [reply]
Thanks I get that. I wanna abstract the notification instance. Does Antd export ArgProps so my wrapper method can correctly type the parameter? I don’t want to have to clone the ArgProps definition 🙁

**Reply by kiner-tang** [reply]
You can get the ArgsProps type like this:

```typescript
type ArgsProps = Parameters<typeof notification.open>[0];

const argsProps: ArgsProps = {

};
```

**Reply by AsuraKev** [reply]
Awesome thanks a lot!

**Comment by kiner-tang** [comment]
> You can get the ArgsProps type like this:
> 
> ```ts
> type ArgsProps = Parameters<typeof notification.open>[0];
> 
> const argsProps: ArgsProps = {
> 
> };
> ```

Maybe I can export this ArgsProps as NotificationArgsProps, but until then, you can use it like this.

**Reply by kiner-tang** [reply, marked as answer]
- #45147
